source/Octopus.Cli/Util/FeatureDetectionExtensions.cs

```diff
@@ -30,8 +30,17 @@ public static async Task<bool> SupportsTenants(this IOctopusAsyncRepository repo
         public static bool UsePostForChannelVersionRuleTest(this RootResource source)
         {
             // Assume octo 3.4 should use the OctopusServer 3.4 POST, otherwise if we're certain this is an older Octopus Server use the GET method
-            SemanticVersion octopusServerVersion;
-            return source == null || !SemanticVersion.TryParse(source.Version, out octopusServerVersion) || octopusServerVersion >= new SemanticVersion("3.4");
+            return source == null || 
+                !SemanticVersion.TryParse(source.Version, out var octopusServerVersion) || 
+                octopusServerVersion >= new SemanticVersion("3.4");
+        }
+        
+        public static bool HasProjectDeploymentSettingsSeparation(this RootResource source)
```

Comment:
Although the `DeploymentSettings` endpoint _technically_ exists prior to 2021.2, this is a check for forwards compatibility to cater to a future where this is the _only_ way to get settings so the specific version it was first provided is less important since the server itself is (currently) backwards compatable.